russh/src/server/session.rs

```diff
@@ -500,6 +500,14 @@ impl Session {
                 .map_err(crate::Error::from)?;
             self.common.write_buffer.buffer.clear();
 
+            if self.common.received_data {
+                // Reset the number of failed keepalive attempts. We don't
+                // bother detecting keepalive response messages specifically
+                // (OpenSSH_9.6p1 responds with REQUEST_FAILURE aka 82). Instead
+                // we assume that the client is still alive if we receive any
+                // data from it.
+                self.common.alive_timeouts = 0;
+            }
```

Comment:
Good catch. This ought to be present on the client as well.

Reply:
Roger that, similar changes applied to the client code in https://github.com/warp-tech/russh/pull/243/commits/ab9d6c74c70415c5c7befb2c16857fe301f42fe1